src/main/java/com/github/teamfusion/spyglassplus/core/mixin/SpyglassItemMixin.java

```diff
@@ -57,41 +57,35 @@ public void setCommanded(boolean commanded) {
 		this.isCommanded = commanded;
 	}
 
-
-	public void setInitiallyCommanded(boolean initiallyCommanded) {
-		this.initiallyCommanded = initiallyCommanded;
-	}
-
 	@Override
 	public void onUseTick(Level level, LivingEntity user, ItemStack stack, int remainingUseTicks) {
 		if (user instanceof Player) {
 			boolean flag = ((ISpyable) user).getSpyGlassStands() != null && !((ISpyable) user).getSpyGlassStands().getSpyGlass().isEmpty();
 
 			ItemStack itemstack2 = flag ? ((ISpyable) user).getSpyGlassStands().getSpyGlass() : user.getUseItem();
 
-
-			int i = EnchantmentHelper.getItemEnchantmentLevel(SpyglassPlusEnchantments.ILLUMINATING.get(), itemstack2);
-			if (i > 0) {
-				user.addEffect(new MobEffectInstance(MobEffects.NIGHT_VISION, 1, 1, false, false, false));
-			}
-			int j = EnchantmentHelper.getItemEnchantmentLevel(SpyglassPlusEnchantments.INDICATING.get(), itemstack2);
-			if (j > 0) {
-				Entity entity = checkEntity(flag ? ((ISpyable) user).getSpyGlassStands() : user, 64.0D);
-				MobEffectInstance effectInstance = new MobEffectInstance(MobEffects.GLOWING, 2, 1, false, false, false);
-				if (entity instanceof LivingEntity) {
-					((LivingEntity) entity).addEffect(effectInstance);
+			if (!level.isClientSide) {
+				int i = EnchantmentHelper.getItemEnchantmentLevel(SpyglassPlusEnchantments.ILLUMINATING.get(), itemstack2);
+				if (i > 0) {
+					user.addEffect(new MobEffectInstance(MobEffects.NIGHT_VISION, 4, 1, false, false, false));
 				}
-			}
-			int k = EnchantmentHelper.getItemEnchantmentLevel(SpyglassPlusEnchantments.COMMAND.get(), stack);
-			if (k > 0) {
-				Entity entity = checkEntityWithNoBlockClip(flag ? ((ISpyable) user).getSpyGlassStands() : user, 64.0D);
-				MobEffectInstance effectInstance = new MobEffectInstance(MobEffects.GLOWING, 2, 1, false, false, false);
-				if (entity instanceof LivingEntity) {
-					((LivingEntity) entity).addEffect(effectInstance);
-					this.setInitiallyCommanded(true);
+				int j = EnchantmentHelper.getItemEnchantmentLevel(SpyglassPlusEnchantments.INDICATING.get(), itemstack2);
+				if (j > 0) {
+					Entity entity = checkEntity(flag ? ((ISpyable) user).getSpyGlassStands() : user, 64.0D);
+					MobEffectInstance effectInstance = new MobEffectInstance(MobEffects.GLOWING, 2, 1, false, false, false);
+					if (entity instanceof LivingEntity) {
+						((LivingEntity) entity).addEffect(effectInstance);
+					}
+				}
+				int k = EnchantmentHelper.getItemEnchantmentLevel(SpyglassPlusEnchantments.COMMAND.get(), stack);
+				if (k > 0) {
+					Entity entity = checkEntityWithNoBlockClip(flag ? ((ISpyable) user).getSpyGlassStands() : user, 64.0D);
+					MobEffectInstance effectInstance = new MobEffectInstance(MobEffects.GLOWING, 2, 1, false, false, false);
+					if (entity instanceof LivingEntity) {
+						((LivingEntity) entity).addEffect(effectInstance);
+					}
```

Comment:
These both seem to do the same thing. You could probable do `if (j > 0 || k > 0) ...`

Reply:
uh have you see the checkEntityWithNoBlockClip and checkEntity method?
I do this because they are separate

Reply:
Ah I missed that! My bad!